# COGS 108 - Data Checkpoint

## Authors

- Andrew Li: Conceptualization, Background research, Writing - original draft, Software
- Roland Wen:  Conceptualization, Methodology, Writing - original draft, Software
- Krasimira Magdadaro: Conceptualization, Background research, Writing - original draft
- Jacob Ramierz: Conceptualization, Data curation, Writing - original draft,  Writing – review & editing
- Zora Dinh: Conceptualization, Methodology, Writing - original draft, Software

## Research Question

Has the production of Slice of Life anime passed Shounen anime from 1990 to 2025 and if so when did this trend start? The trend is defined as the annual ratio of genre-specific releases worldwide to total releases.

## Background and Prior Work

Popular media is a reflection of the interests of the majority and changes over time and place through the influence of culture and environment. Trends can arise when certain genres attract a significant following and creators appeal to this interest, leading to increased production in media with common themes. In terms of anime, two genres that have had a consistently large following and interest are shonen and slice-of-life. Many iconic anime fall under the shonen genre such as Naruto and Dragon Ball and are usually characterized by action and fantasy themes. Similarly, figures from slice-of-life anime such as Lucky Star and Love Live have become easily recognizable from their long-standing popularity and are characterized by lighthearted and relatable stories. With shonen and slice-of-life being popular genres in Japanese media for a long time, we wanted to better understand any potential changes in production between anime of these two genres over the years.

Many shonen series first find their start by having a chapter published in the Weekly Shonen Jump magazine before being compiled into a full manga. The target audience of this magazine is usually teen and young adult boys, which is seen through the frequent use of male protagonists who reflect this audience. The article “The Rise and Fall of Weekly Shonen Jump: A Look at the Circulation of Weekly Jump” focuses on the amount of sales of the Weekly Shonen Jump magazine from 1970 to 2005 and takes note of its peak popularity and when it begins to decline <a href="#ref1">[1]</a>. Observing sales of the magazine, it began its height of popularity in 1993 but “…between 1995 and 2005, while other magazines' sales stayed fairly constant, Shonen Jump's circulation decreased dramatically” <a href="#ref1">[1]</a>. At the same time, competing magazines like Weekly Shonen Magazine also found a slight decline in sales around a similar time period <a href="#ref1">[1]</a>. Comparing this change in sales to the titles in the editions of this time, the decrease in sales coincides with the ending of popular series like Dragon Ball <a href="#ref1">[1]</a>. This study was done in 2007 and only observes sales up to 2005. Additionally, the data solely focuses on the popularity of titles being published through this particular magazine, which may not reflect titles in other magazines or other platforms like television.

In “Trends of Fiction in 2000s Japanese Pop Culture” published in the Electronic Journal of Contemporary Japanese Studies, Motoko Tanaka studies the fall of “the fictional age” (kyokō no jidai) and the beginning of “the age of impossibility” zeitgeists in Japan, along with how that change correlated with the rise and fall of certain anime genres popular among teen-young adult Japanese men <a href="#ref2">[2]</a>. Tanaka argues that the 1970–1995 “fictional age,” which was heralded in by the failure of the Anpo and Zenkyōtō student activism movements, was defined by the desire of young Japanese men to “…seek ideals not in political movements and changes in the real world, but rather in imagined settings such as theme parks and in fictional characters” <a href="#ref2">[2]</a>. While the article doesn’t directly mention shonen anime being popular, it can be logically inferred that anime with strong, morally just male protagonists in fantasy settings fit this cultural desire. Tanaka then discusses the fall of the “fictional age” and the beginning of "the age of impossibility,” influenced by catastrophes such as the 1995 Kobe Earthquake, the Aum sarin gas attack, and 9/11, as well as Japan’s ‘lost decade’ <a href="#ref2">[2]</a>. During this period, more optimistic genres such as traditional shonen saw a decline in prominence, while darker genres such as Sekaikei and Sabaibukei (death game narratives) gained attention <a href="#ref2">[2]</a>. Tanaka also notes that, during the late 2000s, Nichijōkei (slice-of-life) anime began to surge in popularity <a href="#ref2">[2]</a>. Although the paper focuses more on cultural analysis than statistical evidence and is limited mainly to the 1990s–2000s, it still identifies clear genre trends and their cultural context, which helped inform our hypothesis.

A more recent study that focused more on data analysis was performed by Research and Markets, with their results being presented in the article “United States Anime Genres Market Analysis Report 2025: Rise of Isekai and Fantasy Genres - Competition, Trends, Forecasts & Opportunities, 2020–2030” <a href="#ref3">[3]</a>. This article summarizes current trends in the anime industry and predicts future trends and growth, with a focus on the American market. It states that the U.S. anime market has grown significantly due to the rise of streaming services, merchandise growth, and expanding demographic appeal, with the market expected to double by 2030 <a href="#ref3">[3]</a>. It also states that, while shonen and slice-of-life anime continue to be extremely popular genres, they are no longer at the very top. Instead, fantasy and particularly isekai genres are now dominant, influenced by the success of series like Sword Art Online <a href="#ref3">[3]</a>. While the article provides market growth data and industry projections, it does not provide detailed genre-specific statistics and is limited to the 2020s–2030s timeframe.

In general, it seems that, compared to our planned study, most previous studies are either narrower in their temporal focus (we plan on analyzing trends from the 1960s to the 2020s) and/or do not include explicit long-term quantitative analyses and visualizations.


<a name="ref1"></a>[1] The Rise and Fall of Weekly Shonen Jump: A Look at the Circulation of Weekly Jump.
(Article referenced on magazine circulation trends.)

<a name="ref2"></a>[2] Tanaka, M. Trends of Fiction in 2000s Japanese Pop Culture. Electronic Journal of Contemporary Japanese Studies.
https://www.japanesestudies.org.uk/ejcjs/vol14/iss2/tanaka.html

<a name="ref3"></a>[3] United States Anime Genres Market Analysis Report 2025. Research and Markets.
https://www.globenewswire.com/news-release/2025/02/20/3029802/28124/en/United-States-Anime-Genres-Market-Analysis-Report-2025-Rise-of-Isekai-and-Fantasy-Genres-Competition-Trends-Forecasts-Opportunities-2020-2030.html

## Hypothesis


We believe that Slice of Life anime production experienced explosive growth while Shounen production experienced a plateau/drop starting around the year 2000. Shounen was extremely popular around 1990 to the early 2000s. However, starting in the year 2000 we believe that simpler storylines that follow easily digestible characters and lack action have become more popular. We expect to observe an inverse relationship between the production rate of Shounen and Slice of Life worldwide with an intersection point around the year 2000.

## Data

### Ideal Data

**Variables Needed:**
* Anime Genre(s)
  - Used to compare Slice of life and Shonen Data 
* Release Date
  - Used to compare genres through release date
* Date finished airing
  - Used to find length between release and last episode aired
* Duration
  - Used to filter out media less than 10 minutes (Music, Ona, etc...)
* ID
  - Used to identify different anime intuitively
* Media Format (Tv,Movie, etc...)
  - Used for correlation between media type and succes as well as filteration purposes
* Episode Count (if episodic)
  - Used to quantify popularity of production by a studio
  - An anime with more episodes could signify good genre sentiment and expected success

**Variables that could be helpful for further investigation:**
* MAL Score
  - Could be used to see viewer sentiment
* Source
  - Could be used to see if source has a relation to genre succuess
  
**How many observations:**

   - The amount of total observations should be at least 5 times as many as the total shonen and slice of life anime entries in order to get valid proportions.

**Data collection**
	
   - Myanimelist has a large sum of anime related data, a dataset that pulls from myanimelist.net would be the most reliable and robust. While any type of csv       format would be ideal, data hosted on github would be best for easy updating throughout the progression of the project.
     

**Stores/Organized:**

   - The most ideal format for this data would be .csv as it is easily imported into spreadsheet softwares and pandas for better visualization and cleaning
     
   
**Potential Data sets:**

--------------------------------------------------------------

https://www.kaggle.com/datasets/hernan4444/anime-recommendation-database-2020
<span style="color:red">(Dataset not being used)</span>

Pros
* Has release date, Genre data, number of episodes,date finished and mal score
* CSV format 
* Has 17,000+ entries
  
Cons
* Requires Kaggle sign in 
* Data from 2020 (out of date)

--------------------------------------------------------------

https://docs.google.com/spreadsheets/d/1A33miTSU1NHdVO7F8kMT68wgkM6uW8YThKPEsDvLKsc/edit?pli=1&gid=0#gid=0 
  <span style="color:red">(Dataset not being used)</span>

Pros
* Includes revenue information (could be used to relate genre to success(monetary)) 
* CSV format 
* No sign in required
  
Cons
* Doesn’t include genre
* Only 2,500 entries 
* Stopped being updated in 2023


--------------------------------------------------------------

https://github.com/LeoRigasaki/Anime-dataset 
<span style="color:green">(Dataset Chosen by group for analysis)</span>

Pros
* Has release date, Genre data, number of episodes,date finished and mal score
* CSV format 
* GitHub hosted 
* Has 21,000+ entries
* Ratio of Shounen and Slice of life entries to total entries is around 1:6
* Updated daily
  
Cons
* Many useless columns (extra cleaning)


### Data overview

**Dataset Name:** anime_seasonal_2026

**Link to the dataset:** https://github.com/LeoRigasaki/Anime-dataset/tree/main/data/raw

**Number of observations:** 21784

**Number of variables:** 29

**Description of the variables most relevant to this project:**

* anime_id
    - Identification number for anime 
* title
    - Name of the anime 
* type
    - Type of media (tv,movie,ona,cm,music,etc…)
* episodes
    - Total episode count in given season
* duration
    - Duration of anime in seconds
* start_date
    - The date in which the anime was released or started airing 
* end_date
    - The date in which the anime finished airing 
* genres
    - A list of relevant genres for the given anime

**Short Comings:**
* The data set has a large number of variables that are not useful for our analysis
* The dataset includes tv,movie,music,unknown,special,tv_special,ova,pv,cm, ona and unknown under the types column if not handled properly this could skew our results.
* There are quite a few entries with duration set as 0 and episodes set as 0 this will result in need for extra cleaning and loss of data entries.


In [2]:
# Run this code every time when you're actively developing modules in .py files.  It's not needed if you aren't making modules
#
## this code is necessary for making sure that any modules we load are updated here 
## when their source code .py files are modified

%load_ext autoreload
%autoreload 2

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


In [3]:
# Setup code -- this only needs to be run once after cloning the repo!
# this code downloads the data from its source to the `data/00-raw/` directory
# if the data hasn't updated you don't need to do this again!

# if you don't already have these packages (you should!) uncomment this line
# %pip install requests tqdm

import sys
sys.path.append('./modules') # this tells python where to look for modules to import

import clean_data

import pandas as pd
import seaborn as sns

# We uploaded our dataset on github directly as the original source looks for new anime and update the dataset every day.

### Anime Production Dataset (1969~now)

For this anime dataset, we are focusing on the columns ‘anime_id’, ‘title’, ‘type’, ‘episodes’, ‘duration’, ‘genres’, ‘start_date’, and ‘end_date’. The ‘title’ column shows the official untranslated titles of each series in this dataset and romanized the Japanese titles if applicable. In the ‘duration’ column, the length of each episode is listed in seconds as an integer. Titles categorized as “TV” under ‘type’ are usually between 720-1440 seconds in length while movies tend to be 3600 seconds or longer. Under the ‘genres’ column, each title is categorized with a list of all the genres that are applicable. Titles can have at least one or more genres in its list. We will only be comparing titles where either the genres shonen or slice-of-life appear with no overlap. The ‘start_date’ column lists the initial date that the title began to air in year-month-day format. The ‘end_date’ is formatted similarly with a date in year-month-day format. A title will not have a value under this column if the series is ongoing and has not been finished airing.

The data in this dataset is updated daily from the website MyAnimeList which was first created in 2004. This website is mostly used by English-speaking users and is paired with streaming services popular in the U.S. like Crunchyroll and Hidive. As a result, the availability and data regarding older titles prior to the website’s creation may be less available. Additionally, user based data may tend to reflect Western audiences who use the website itself as well as the streaming services partnered with them while leaving out titles that are only available in Japan.

#### Load the dataset

In [4]:
# load the dataset
df = pd.read_csv('./data/00-raw/anime_data.csv')
df.head()

anime_id                                  title english_title  \
0      3009                             Tiger Mask    Tiger Mask   
1      2830           Chibikko Remi to Meiken Kapi           NaN   
2      8899                  Kaitei Shounen Marine    Marine Boy   
3     12789                          Yasashii Lion           NaN   
4     22221  Monoshiri Daigaku: Ashita no Calendar           NaN   

    japanese_title   type  episodes  duration           status        source  \
0          タイガーマスク     tv       105      1380  finished_airing         manga   
1      ちびっ子レミと名犬カピ  movie         1      4860  finished_airing         novel   
2          海底少年マリン     tv        65      1500  finished_airing           NaN   
3         やさしいライオン  movie         1      1620  finished_airing  picture_book   
4  ものしり大学　明日のカレンダー     tv      1274       300  finished_airing      original   

        season  ... watching completed on_hold  dropped  plan_to_watch  \
0    fall 1969  ...        0         0       0        0              0   
1  winter 1970  ...        0         0       0        0              0   
2  winter 1969  ...        0         0       0        0              0   
3  winter 1970  ...        0         0       0        0              0   
4  summer 1966  ...        0         0       0        0              0   

   start_date    end_date  broadcast_day  broadcast_time  \
0  1969-10-02  1971-09-30            NaN             NaN   
1  1970-03-17  1970-03-17            NaN             NaN   
2  1969-01-13  1971-09-02          other           17:00   
3  1970-03-21  1970-03-21            NaN             NaN   
4  1966-07-01  1970-08-02          other           18:55   

                                            synopsis  
0  Tiger Mask (whose real name was Naoto Date) wa...  
1  Based on Sans Famille by Hector Malot.  This i...  
2  Yet another expansion of the Marine Boy series...  
3  This is an animated version of Yanase Takashi'...  
4  Another continuation of Instant History with a...  

[5 rows x 29 columns]

#### The dataset is already tidy
We started by finding if there are any duplicates of anime titles in different rows. We found that there are two rows with the exact same title "Moonlight". 

In [5]:
all_dupes = df[df.duplicated(subset=['title'], keep=False)].reset_index()
all_dupes

index  anime_id      title english_title japanese_title   type  episodes  \
0  20161     59994  Moonlight     Moonlight         ムーンライト  music         1   
1  20530     60524  Moonlight           NaN            NaN  music         1   

   duration           status    source  ... watching completed on_hold  \
0       205  finished_airing  original  ...        0         0       0   
1       203  finished_airing  original  ...        0         0       0   

  dropped  plan_to_watch  start_date    end_date  broadcast_day  \
0       0              0  2024-09-30  2024-09-30            NaN   
1       0              0  2024-12-09  2024-12-09            NaN   

   broadcast_time                                           synopsis  
0             NaN  Music video for the song Moonlight by Suisei H...  
1             NaN  Music video for the song Moonlight by MOON SHARK.  

[2 rows x 30 columns]

In [6]:
print(all_dupes.loc[0, 'synopsis'])
print(all_dupes.loc[1, 'synopsis'])

Music video for the song Moonlight by Suisei Hoshimachi.
Music video for the song Moonlight by MOON SHARK.


After some research, we found out that they are different music videos that happen to have the exact same title. The duration and start/end date are also different. Thus, the dataset is already tidy as each row represents one data point (one single anime).
We will discuss how we process different media types later.

#### Dataset size
The original dataset contains 21,784 entries and 29 columns. Out of the 29 columns, we will be only using 8 of them as described in the dataset description. There are mainly 5 media types: tv (tv, tv_special, ova, ona), movie (movie), music (music), pv (pv, cm), and others (unknown, special).

In [7]:
total_unfiltered_entries = df.shape[0]
print(f'There are {total_unfiltered_entries} entries in the raw dataset')

There are 21784 entries in the raw dataset


In [8]:
print(f"The columns being used are {clean_data.USEFUL_COLS}")
df = clean_data.filter_cols(df=df, columns_to_keep=clean_data.USEFUL_COLS)
print("Note that status is only used for determining whether an anime is airing currently.")

The columns being used are ['anime_id', 'title', 'type', 'episodes', 'duration', 'genres', 'start_date', 'end_date', 'status']
Note that status is only used for determining whether an anime is airing currently.


#### Missing data
We start by analyzing how many entries contain at least 1 NA.

In [9]:
rows_with_na = df.isna().any(axis=1)
print(f'There are {rows_with_na.sum()} rows with at least 1 NA which is {round(rows_with_na.sum()/df.shape[0]*100, 1)}% of the dataset')

There are 537 rows with at least 1 NA which is 2.5% of the dataset


In [10]:
for col in df.columns:
    print(f'{df[rows_with_na][col].isna().sum()} rows have no {col}')

0 rows have no anime_id
0 rows have no title
0 rows have no type
0 rows have no episodes
0 rows have no duration
25 rows have no genres
0 rows have no start_date
513 rows have no end_date
0 rows have no status


The only data missing are genres and end date. Since genres is critical for our analysis and there are only 25 rows without it, we will drop rows whose genres is NA.

In [11]:
df = df.dropna(subset=['genres'])

Since 513 entries do not have an end date, we will focus our analysis on start date mainly.

#### Outliers

In [12]:
df.describe()

anime_id      episodes      duration
count  21759.00000  21759.000000  21759.000000
mean   33518.03024     11.792959   1342.375155
std    20318.11011     40.899307   1591.492333
min        1.00000      0.000000      0.000000
25%    12864.00000      1.000000    219.000000
50%    36862.00000      1.000000    920.000000
75%    52273.50000     12.000000   1481.000000
max    63469.00000   1818.000000  10081.000000

There are some outliers in the amount of episodes and duration. The maximum duration is 10081 seconds or around 2 hours and 50 minutes. This is not especially long for a movie. Thus, we will focus on the outliers of episodes only.

In [13]:
df.sort_values(by='episodes', ascending=False).head(10)

anime_id                                  title type  episodes  \
2552     12393                             Oyako Club   tv      1818   
399       2471                        Doraemon (1979)   tv      1787   
9762     23539                               Gudetama   tv      1717   
194      23349                 Kirin Monoshiri Yakata   tv      1565   
232       6277     Manga Nippon Mukashibanashi (1976)   tv      1471   
263       8213                        Hokahoka Kazoku   tv      1428   
456      32448               Kirin Ashita no Calendar   tv      1306   
4        22221  Monoshiri Daigaku: Ashita no Calendar   tv      1274   
78       10241                  Sekai Monoshiri Ryoko   tv      1006   
1185     20117                         Kotowaza House   tv       773   

      duration                                             genres  start_date  \
2552       300                                             Comedy  1994-10-03   
399        660  Adventure;Anthropomorphic;Comedy;Fantasy;Kids;...  1979-04-02   
9762        80                                Comedy;Gourmet;Kids  2014-03-31   
194        300                                               Kids  1975-01-01   
232       1500                                 Fantasy;Historical  1976-01-07   
263        300                                             Comedy  1976-10-01   
456        300                                    Historical;Kids  1980-01-01   
4          300                                         Historical  1966-07-01   
78         300                                             Comedy  1971-10-01   
1185       300                                             Comedy  1987-04-04   

        end_date           status  
2552  2013-03-30  finished_airing  
399   2005-03-18  finished_airing  
9762  2023-12-24  finished_airing  
194   1979-12-31  finished_airing  
232   1994-09-03  finished_airing  
263   1982-03-31  finished_airing  
456   1984-10-06  finished_airing  
4     1970-08-02  finished_airing  
78    1974-12-31  finished_airing  
1185  1994-09-28  finished_airing

The anime with a lot of episodes usually air across multiple years. In our analysis, we should also take that into account since it's an indication of how important those anime are to the industry.

#### Cleaning the dataset
We clean the dataset in 3 steps:
1. Process genres
2. Process dates
3. Process type/duration

##### Process genres
The genres column is in the form "genre 1;genre 2;genre 3;..." Since we only care about the slice-of-life and shounen anime, we create two binary columns "is_slice_of_life" and "is_shounen" from the genres column. If "Slice of life" is contained the genre, we will set is_slice_of_life to True, otherwise, False. If "Shounen" is contained the genre, we will set is_shounen to True, otherwise, False. There are a few anime that are both slice-of-life and shounen. Since there are only a few of them, we remove them from our analysis.

In [14]:
genres_to_analyze = (clean_data.SHOUNEN_TAG, clean_data.SLICE_OF_LIFE_TAG)
df = clean_data.gen_binary_categories(df=df, tags_to_check=genres_to_analyze)
slice_of_life_and_shounen_cnt = df[df['is_slice_of_life'] & df['is_shounen']].shape[0]
print(f"There are {slice_of_life_and_shounen_cnt} ({round(slice_of_life_and_shounen_cnt/df.shape[0], 2)}%) entries that are both slice of life and shounen")

There are 31 (0.0%) entries that are both slice of life and shounen


In [15]:
df = clean_data.filter_multicategory_anime(df=df, tags_to_check=genres_to_analyze)
df.head(5)

anime_id                                  title   type  episodes  duration  \
0      3009                             Tiger Mask     tv       105      1380   
1      2830           Chibikko Remi to Meiken Kapi  movie         1      4860   
2      8899                  Kaitei Shounen Marine     tv        65      1500   
3     12789                          Yasashii Lion  movie         1      1620   
4     22221  Monoshiri Daigaku: Ashita no Calendar     tv      1274       300   

                               genres  start_date    end_date  \
0  Combat Sports;Drama;Shounen;Sports  1969-10-02  1971-09-30   
1                     Adventure;Drama  1970-03-17  1970-03-17   
2        Action;Adventure;Kids;Sci-Fi  1969-01-13  1971-09-02   
3    Award Winning;Fantasy;Kids;Music  1970-03-21  1970-03-21   
4                          Historical  1966-07-01  1970-08-02   

            status  is_shounen  is_slice_of_life  
0  finished_airing        True             False  
1  finished_airing       False             False  
2  finished_airing       False             False  
3  finished_airing       False             False  
4  finished_airing       False             False

##### Process dates
We convert the start and end dates from string to datetime objects. Then, we added start_year, end_year, and aired_days as floats since we will mainly analysis the production trend across years. For shows that are still airing, we calculate its end year and aired day by using ref_data (2026-02-01 for this analysis) as its last aired date.

In [16]:
df = clean_data.add_airing_date_features(df=df)
df.head(5)

anime_id                                  title   type  episodes  duration  \
0      3009                             Tiger Mask     tv       105      1380   
1      2830           Chibikko Remi to Meiken Kapi  movie         1      4860   
2      8899                  Kaitei Shounen Marine     tv        65      1500   
3     12789                          Yasashii Lion  movie         1      1620   
4     22221  Monoshiri Daigaku: Ashita no Calendar     tv      1274       300   

                               genres start_date   end_date           status  \
0  Combat Sports;Drama;Shounen;Sports 1969-10-02 1971-09-30  finished_airing   
1                     Adventure;Drama 1970-03-17 1970-03-17  finished_airing   
2        Action;Adventure;Kids;Sci-Fi 1969-01-13 1971-09-02  finished_airing   
3    Award Winning;Fantasy;Kids;Music 1970-03-21 1970-03-21  finished_airing   
4                          Historical 1966-07-01 1970-08-02  finished_airing   

   is_shounen  is_slice_of_life  start_year  end_year  aired_days  
0        True             False      1969.0    1971.0       729.0  
1       False             False      1970.0    1970.0         1.0  
2       False             False      1969.0    1971.0       963.0  
3       False             False      1970.0    1970.0         1.0  
4       False             False      1966.0    1970.0      1494.0

##### Process type/duration
The duration in this dataset is stored in seconds. For a better way of gauging the duration of every anime, we convert the duration column from seconds to minutes. Additionally, to filter out media with a very short run time, we filter out all rows with a duration of less than 10 minutes.

In [17]:
df = clean_data.seconds_to_mins(df, 'duration')
df = df[df['duration'] >= 10]
df.head(5)

anime_id                             title   type  episodes  duration  \
0      3009                        Tiger Mask     tv       105        23   
1      2830      Chibikko Remi to Meiken Kapi  movie         1        81   
2      8899             Kaitei Shounen Marine     tv        65        25   
3     12789                     Yasashii Lion  movie         1        27   
5     17445  Kyojin no Hoshi: Dai League Ball  movie         1        70   

                               genres start_date   end_date           status  \
0  Combat Sports;Drama;Shounen;Sports 1969-10-02 1971-09-30  finished_airing   
1                     Adventure;Drama 1970-03-17 1970-03-17  finished_airing   
2        Action;Adventure;Kids;Sci-Fi 1969-01-13 1971-09-02  finished_airing   
3    Award Winning;Fantasy;Kids;Music 1970-03-21 1970-03-21  finished_airing   
5            Drama;Sports;Team Sports 1970-03-21 1970-03-21  finished_airing   

   is_shounen  is_slice_of_life  start_year  end_year  aired_days  
0        True             False      1969.0    1971.0       729.0  
1       False             False      1970.0    1970.0         1.0  
2       False             False      1969.0    1971.0       963.0  
3       False             False      1970.0    1970.0         1.0  
5       False             False      1970.0    1970.0         1.0

##### Save the preprocessed dataset

In [18]:
df.to_csv('data/02-processed/preprocessed_anime_data.csv', index=False)

## Ethics

### A. Data Collection
 - [X] **A.1 Informed consent**: If there are human subjects, have they given informed consent, where subjects affirmatively opt-in and have a clear understanding of the data used to which they consent?
 - [X] **A.2 Collection bias**: Have we considered sources of bias that could be introduced during data collection and survey design and taken steps to mitigate those?
> Older anime are less likely to be recorded in modern dataset, especially for more niche anime. The genre tagging process could also be biased as there are no purely objective definitions of the genres.
 - [X] **A.3 Limit PII exposure**: Have we considered ways to minimize exposure of personally identifiable information (PII) for example through anonymization or not collecting information that isn't relevant for analysis?
 - [X] **A.4 Downstream bias mitigation**: Have we considered ways to enable testing downstream results for biased outcomes (e.g., collecting data on protected group status like race or gender)?

### B. Data Storage
 - [X] **B.1 Data security**: Do we have a plan to protect and secure data (e.g., encryption at rest and in transit, access controls on internal users and third parties, access logs, and up-to-date software)?
> Even though our dataset does not include any confidential data, we will ensure data integrity and up-to-date software. We need data integrity and up-to-date software to enable reproducibility of our experiments.
 - [X] **B.2 Right to be forgotten**: Do we have a mechanism through which an individual can request their personal information be removed?
 - [X] **B.3 Data retention plan**: Is there a schedule or plan to delete the data after it is no longer needed?
> No, since we do not have ownership of the dataset, we cannot delete the dataset.

### C. Analysis
 - [X] **C.1 Missing perspectives**: Have we sought to address blindspots in the analysis through engagement with relevant stakeholders (e.g., checking assumptions and discussing implications with affected communities and subject matter experts)?
> Although direct engagement with production studios is not feasible, we sought to address blindspots by reviewing community tagging documentation to fully understand the possible bias behind genre tagging.
 - [X] **C.2 Dataset bias**: Have we examined the data for possible sources of bias and taken steps to mitigate or address these biases (e.g., stereotype perpetuation, confirmation bias, imbalanced classes, or omitted confounding variables)?
> Older anime are less likely to be recorded in modern dataset, especially for more niche anime. The genre tagging process could also be biased as there are no purely objective definitions of the genres. We admit these biases, but they are beyond our ability to address. If we were to address them, we would ask different people to vote on the genre of each anime. Since there are more than 10,000 anime in the dataset, such an approach would not be feasible for the timeline of this project.
 - [X] **C.3 Honest representation**: Are our visualizations, summary statistics, and reports designed to honestly represent the underlying data?
> We recognize that it is nearly impossible to have a completely unbiased representation of the underlying data. We will cross check each others’ works to ensure that a minimum amount of bias is achieved throughout the project. We will also note unresolvable biases that may occur in our analysis in the report.
 - [X] **C.4 Privacy in analysis**: Have we ensured that data with PII are not used or displayed unless necessary for the analysis?
 - [X] **C.5 Auditability**: Is the process of generating the analysis well documented and reproducible if we discover issues in the future?
> Yes, we will document all the details of our analysis to ensure its auditability. We will also use a fixed timeframe (eg. 1990~2025) to avoid ambiguity in the dataset as it is being updated daily to include the newest anime.

### D. Modeling
 - [X] **D.1 Proxy discrimination**: Have we ensured that the model does not rely on variables or proxies for variables that are unfairly discriminatory?
> Though the dataset is being analyzed from a non-Japanese perspective, the data itself is largely representative of English-speaking anime audience as a whole, and thus we want our analysis to focus on that and limit anything that would be particularly discriminatory (ie specific regions or areas)
 - [X] **D.2 Fairness across groups**: Have we tested model results for fairness with respect to different affected groups (e.g., tested for disparate error rates)?
> We are factoring in disparate error via limiting the amount of ‘groups’ we are analyzing, as our model largely wants to hone in on similarly sized, comparable media genres for best prediction
 - [X] **D.3 Metric selection**: Have we considered the effects of optimizing for our defined metrics and considered additional metrics?
 - [X] **D.4 Explainability**: Can we explain in understandable terms a decision the model made in cases where a justification is needed?
 - [X] **D.5 Communicate limitations**: Have we communicated the shortcomings, limitations, and biases of the model to relevant stakeholders in ways that can be generally understood?
> We understand that the model is training on Japanese media using English-speaking audiences and their data, and thus may not be the most optimal predictor of what Japanese audiences and companies’ decisions end up being

### E. Deployment
 - [X] **E.1 Monitoring and evaluation**: Do we have a clear plan to monitor the model and its impacts after it is deployed (e.g., performance monitoring, regular audit of sample predictions, human review of high-stakes decisions, reviewing downstream impacts of errors or low-confidence decisions, testing for concept drift)?
> Monitoring the model over time according to future trends compared to our fixed timeframe (as rating is largely skewed by modern audiences’ tastes compared to more dated media) and checking if the conclusions made now still apply in the future
 - [X] **E.2 Redress**: Have we discussed with our organization a plan for response if users are harmed by the results (e.g., how does the data science team evaluate these cases and update analysis and models to prevent future harm)?
 - [X] **E.3 Roll back**: Is there a way to turn off or roll back the model in production if necessary?
 - [X] **E.4 Unintended use**: Have we taken steps to identify and prevent unintended uses and abuse of the model and do we have a plan to monitor these once the model is deployed?
> Properly outlining/defining what the model is primarily for would be the biggest step in preventing unintended usage, especially as a policy or disclaimer for usage


## Team Expectations 

* Communication will primarily be done using Discord (we have our own server)
* Usual tone when discussing things should be casual, polite, and friendly unless something serious comes up (like a divisive conflict in opinion between members), in which case politeness should still be maintained
* Decisions should be made unanimously by the group
* Each person will focus on a specific task/aspect of the project (different parts of the writing, EDA, preprocessing, visualizations, etc.), with tasks only being assigned with the person’s full knowledge and consent. However, it is expected for everyone to collaborate and keep in touch with each other’s work so that everyone has a full picture of what is happening. 
* Keep an updated and fair project timeline so that everyone knows what they are doing
* If a person ends up not being able to do a task or wants to do something else, the group can arrange a swap of who does what. This will probably be arranged in Discord and people can bring up their issues during either a weekly meeting or by just typing in the chat. The person with the issue should contact the group ASAP. The group can have one of the other members complete the work for the person with the issue, while the person with the issue will be assigned more work in some other part of the project to make sure they are still doing their part.
* If one or more members believe that someone isn’t putting in the work that they should be doing, they should try and make their concerns heard by the problematic member via either a private message on Discord or Email. If the issues persist or if the problematic member becomes/continues to be more difficult to work with, the member should inform the rest of the group and from there they can decide what to do about the problem.


## Project Timeline Proposal

| Meeting Date | Meeting Time | Completed Before Meeting | Discuss at Meeting |
|--------------|--------------|--------------------------|--------------------|
| Feb 4 | 7:00–8:00pm | All sections of the project proposal done | Figure out who does the preprocessing and early EDA for the code |
| Feb 6 | 3:30–4:30pm | Beginning written sections (background, data, ethics) polished up, some preprocessing done | Check in on progress for preprocessing, maybe some very early data analysis |
| Feb 13 | 3:30–4:30pm | More preprocessing and some EDA completed | Check in on progress for preprocessing|
| Feb 18 (Data checkpoint) | 7:00–8:00pm | Preprocessing done| Turn in data checkpoint, discuss who does what major EDA and discuss who will do what for the main analyses and visualizations |
| Feb 20 | 3:30–4:30pm | Complete most of the EDA, some work done on main analyses and visualizations | Check in on EDA progress and main analyses + visualizations |
| Feb 27 | 3:30–4:30pm | Have EDA be done, along with most main analyses and visualizations | Check in on main analyses + visualizations progress, discuss who does what for the final written parts (conclusion + discussion, limitations, next steps, final team member contributions) |
| Mar 4 (EDA checkpoint) | 7:00–8:00pm | Completed main analyses and visualizations, along with most of the final written parts | Check in on final written part progress, discuss how to film the project video |
| Mar 6 | 3:30–4:30pm | Have the final parts of the project be completed (everything done) | Check in to do one last sweep through all the parts of the project, organize who will turn it in |
| Mar 13 | 3:30–4:30pm | Empty — reserved in case extra meetings are needed | Empty — reserved in case extra meetings are needed |
| Mar 18 (everything done) | 7:00–8:00pm | Empty — reserved in case extra meetings are needed | Empty — reserved in case extra meetings are needed |
